src/CBLURLEndpointListener_Internal.hh

```diff
@@ -0,0 +1,157 @@
+//
+//  CBLURLEndpointListener_Internal.hh
+//
+// Copyright © 2025 Couchbase. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#pragma once
+
+#include "Internal.hh"
+#include "CBLCollection_Internal.hh"
+#include "CBLDatabase_Internal.hh"
+#include "c4.h"
+
+#ifdef COUCHBASE_ENTERPRISE
+
+CBL_ASSUME_NONNULL_BEGIN
+
+struct CBLListenerAuthenticator {
+    
+};
+
+struct CBLURLEndpointListener final : public CBLRefCounted {
+public:
+    static CBLURLEndpointListener* _cbl_nullable create(const CBLURLEndpointListenerConfiguration &conf, CBLError* _cbl_nullable outError) {
+        if (conf.collectionCount == 0) {
+            if (outError) {
+                outError->domain = kCBLDomain;
+                outError->code = kCBLErrorInvalidParameter;
+            }
+            return nullptr;
+        }
+        return retain(new CBLURLEndpointListener(conf));
+    }
+
+    const CBLURLEndpointListenerConfiguration* configuration() const { return &_conf; }
+
+    uint16_t port() const {
+        if (_port == 0 && _c4listener) {
+            _port = c4listener_getPort(_c4listener);
+        }
+        return _port;
+    }
+
+    FLMutableArray _cbl_nullable getUrls() const {
+        FLMutableArray ret = (FLMutableArray) nullptr;
+        if (_c4listener) {
+            CBLDatabase* cblDb = CBLCollection_Database(_conf.collections[0]);
+            cblDb->c4db()->useLocked([&](C4Database* db) {
+                ret = c4listener_getURLs(_c4listener, db, kC4SyncAPI, nullptr);
+            });
+        }
+        return ret;
+    }
+
+    CBLConnectionStatus getConnectionStatus() const {
+        unsigned int total = 0, active = 0;
+        if (_c4listener) c4listener_getConnectionStatus(_c4listener, &total, &active);
+        return {total, active};
+    }
+
+    bool start(CBLError* _cbl_nullable outError) {
```

Comment:
At the C++ level, throw exception instead of using out error.

Reply:
done